rls/src/actions/mod.rs

```diff
@@ -281,12 +282,21 @@ impl InitActionContext {
     fn file_edition(&self, file: PathBuf) -> Option<Edition> {
         let files_to_crates = self.file_to_crates.lock().unwrap();
 
-        let editions: HashSet<_> = files_to_crates.get(&file)?.iter().map(|c| c.edition).collect();
+        let editions: HashSet<_> = files_to_crates
+            .get(&file)
+            .map(|crates| crates.iter().map(|c| c.edition).collect())
+            .unwrap_or_default();
 
         let mut iter = editions.into_iter();
         match (iter.next(), iter.next()) {
             (ret @ Some(_), None) => ret,
-            _ => None,
+            (Some(_), Some(_)) => None,
+            _ => {
+                // fall back on checking the root manifest for edition
```

Comment:
nit:
```suggestion
                // fall back on checking the root manifest for package edition
```
To emphasize that we only fall back to looking for package edition (and that it won't work if only a crate target edition is specified)